test/guardian/plug/verify_header_test.exs

```diff
@@ -172,4 +172,96 @@ defmodule Guardian.Plug.VerifyHeaderTest do
     assert conn.status == 401
     refute conn.halted
   end
+
+  describe "with verify_cookie option" do
+    defmodule ImplJwt do
+      @moduledoc false
+
+      use Guardian,
+        otp_app: :guardian,
+        token_module: Guardian.Token.Jwt,
+        issuer: "MyApp",
+        verify_issuer: true,
+        secret_key: "foo-de-fafa",
+        allowed_algos: ["HS512", "ES512"],
+        ttl: {4, :weeks},
+        secret_fetcher: Guardian.Support.TokenModule.SecretFetcher,
+        token_ttl: %{
+          "access" => {1, :day},
+          "refresh" => {2, :weeks}
+        },
+        handler: __MODULE__.Handler
+
+      def subject_for_token(%{id: id}, _claims), do: {:ok, "User:#{id}"}
+      def resource_from_claims(%{"sub" => "User:" <> sub}), do: {:ok, %{id: sub}}
+
+      def the_secret_yo, do: config(:secret_key)
+      def the_secret_yo(val), do: val
+
+      def verify_claims(claims, opts) do
+        if Keyword.get(opts, :fail_owner_verify_claims) do
+          {:error, Keyword.get(opts, :fail_owner_verify_claims)}
+        else
+          {:ok, claims}
+        end
+      end
+
+      def build_claims(claims, _opts) do
+        Map.put(claims, "from_owner", "here")
+      end
+    end
+
+    setup do
+      impl = __MODULE__.ImplJwt
+      handler = __MODULE__.Handler
+      {:ok, token, claims} = __MODULE__.ImplJwt.encode_and_sign(@resource)
+      {:ok, %{claims: claims, conn: conn(:get, "/"), token: token, impl: impl, handler: handler}}
+    end
+
+    test "when session is valid", ctx do
+      conn =
+        :get
+        |> conn("/")
+        |> put_req_header("authorization", ctx.token)
+        |> VerifyHeader.call(module: ctx.impl, verify_cookie: [module: ctx.impl])
```

Comment:
To avoid all the `module: ctx.impl` madness you can use `|> Pipeline.put_module(ctx.impl)` then we can also test the option true `to verify_cookie`, what do you think?

Reply:
I will add a test case for `refresh_from_cookie: true` (new `verify_cookie` :)).